onchain/src/quests/template_quest.cairo

```diff
@@ -0,0 +1,80 @@
+use art_peace::templates::interfaces::{
+    ITemplateStoreDispatcher, ITemplateStoreDispatcherTrait, ITemplateVerifierDispatcher,
+    ITemplateVerifierDispatcherTrait, TemplateMetadata
+};
+
+#[starknet::contract]
+pub mod TemplateQuest {
+    use starknet::{ContractAddress, get_caller_address};
+    use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+    use art_peace::quests::{IQuest, QuestClaimed};
+
+    #[storage]
+    struct Storage {
+        art_peace: IArtPeaceDispatcher,
+        reward: u32,
+        claimed: LegacyMap<ContractAddress, bool>,
+        template_store: ITemplateStoreDispatcher
+    }
+
+    #[derive(Drop, Serde)]
+    pub struct TemplateQuestInitParams {
+        pub art_peace: ContractAddress,
+        pub reward: u32,
+        pub template_store: ContractAddress
```

Comment:
This param isn't needed because `template_store` = `art_peace` address. Since the template store is a component.

Reply:
if template_store is not needed, then art_peace in the constructor will change to something like this 

```
 fn constructor(ref self: ContractState, init_params: TemplateQuestInitParams,) {
        self.reward.write(init_params.reward);
        self
            .art_peace
            .write(ITemplateStoreDispatcher { contract_address: init_params.template_store });
    }
```
art_peace will now carry ITemplateStoreDispatcher,

Reply:
I meant you could do something like 
`self.template_store.write(ITemplateStoreDispatcher { contract_address: init_params.art_peace });`

And use `art_peace: ContractAddress` in storage when checking the art_peace contract address, if that makes sense.

Reply:
Alright, make sense